iree/compiler/Dialect/Flow/Transforms/FoldCompatibleDispatchRegions.cpp

```diff
@@ -229,6 +230,36 @@ bool rhsHasRootOnlyOp(DispatchRegionOp &lhs, DispatchRegionOp &rhs) {
   return false;
 }
 
+// Returns true if lhs has ops that can only be leaf op and will lose the
```

Comment:
Comment is either misleading or off. A root op can be a leaf op (if there is a single op). If you merge something that is a `rootOp` with a dispatch region, its no longer a root op. But if you merge a dispatch region which has a single operation with its "consumer", the operation will still be a leaf. Maybe this needs to be made clearer?

Reply:
I think we can use first/last op for discussion here because it's confusing...

> If you merge something that is a rootOp with a dispatch region, its no longer a root op.

It can still be the first op if the operation still take function inputs as operands.
It can still be the last op if `rhs` doesn't take the result as operand. Otherwise there is a dep.

In this file, `lhs` stands for producer and `rhs` stands for consumer. This is for checking if the result of the last op in `lhs` will be used in `rhs`.